component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -201,6 +209,35 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             if (context.isRetrying()) {
                 retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
             }
+            IdentityErrorMsgContext errorContext = IdentityUtil.getIdentityErrorMsg();
+            IdentityUtil.clearIdentityErrorMsg();
+
+            String errorParam = StringUtils.EMPTY;
+            if (showAuthFailureReason) {
+                if (errorContext != null && errorContext.getErrorCode() != null) {
```

Comment:
for `errorContext.getErrorCode() != null` we can use `StringUtils`

Reply:
Addressed in https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/156